### ISSUE TITLE
Backend-agnostic JIT compilation support for non supervised learning workflows

### ISSUE THREAD

**Issue by neo-alex** [keras-team/keras#19315]
Hello!

Keras 3 provides a very nice backend-agnostic way to enable/disable eager mode & JIT compilation for Supervised Learning workflows: adjusting run_eagerly & jit_compile in `model.compile()` before calling `model.fit()`.

However, for other workflows like Reinforcement Learning where typically the model is called directly (e.g. `model(state)` in [this official example](https://keras.io/examples/rl/actor_critic_cartpole)), it seems that we lose this nice capability - or am I missing something?

It would be great to have a way to support it without having to re-write some boilerplate code. I believe that `model.predict_on_batch()` almost does the job (since it seems to take into account the jit_compile value set in model.compile()), except that it currently doesn't allow to keep a tensor output (instead of casting it to Numpy) and to enable gradient computation.

Any recommendation/idea to make alternative workflows work smoothly with eager/JIT whatever the backend would be very welcome. Thanks!

**Comment by sachinprasadhs** [comment]
I don't think this will give great performance boost for the example you have mentioned when you explicitly enable `JIT`.

If you implement `jit_compile`, it may cause gradient recording issues, you need to be careful while designing.

**Comment by neo-alex** [comment]
Thank you for your reply, but I am not sure to get your point about "gradient recording issues". In my understanding, jit-compilation is the new default in Keras 3 (at least for JAX/TF backends) when training with `model.fit()` -which uses of course gradient descent for learning- and it results in a dramatic training time reduction.

I wish we had the same capability to also enable it in a cross-backend way when just calling the model in a training or inference loop (without using `model.fit()`).
